deluge/ui/gtkui/piecesbar.py

```diff
@@ -124,11 +119,10 @@ def __draw_pieces(self):
             )
             ctx = cairo.Context(self.__pieces_overlay)
             start_pos = 0
-            num_pieces = self.__num_pieces and self.__num_pieces or len(self.__pieces)
-            piece_width = self.__width * 1.0 / num_pieces
+            piece_width = self.__width * 1.0 / (self.__num_pieces or len(self.__pieces))
 
-            for state in self.__pieces:
-                color = self.gtkui_config["pieces_color_%s" % COLOR_STATES[state]]
+            def fill():
+                global start_pos
```

Comment:
Really should avoid using global.. In this situation a good alternative would be to enumerate for loop and pass as an arg to fill function. 

However if you implement the suggestion in my other comment below you can remove this function and put the code back into the for loop.